src/CloudNative.CloudEvents.Amqp/AmqpClientExtensions.cs

```diff
@@ -85,6 +84,17 @@ public static class AmqpClientExtensions
                         cloudEvent[attributeName] = property.Value;
                     }
                 }
+                // Populate the data after the rest of the CloudEvent
+                if (message.BodySection is Data data)
+                {
+                    // TODO: Will data.Binary always be populated? Perhaps it'll be data.Buffer?
+                    formatter.DecodeBinaryModeEventData(data.Binary, cloudEvent);
+                }
+                else if (message.BodySection is object)
+                {
+                    throw new ArgumentException("Binary mode data in AMQP message must be in the application data section");
```

Comment:
Isn't this true for structured mode as well?

Reply:
Yes - we can probably refactor all of this a bit, to be honest.